File: Engine/src/jvmMain/java/org/etieskrill/engine/input/CursorInputAdapter.java
package org.etieskrill.engine.input;

public interface CursorInputAdapter extends CursorInputHandler {

    @Override
    default boolean invokeClick(KeyEvent button, Key.Action action, double posX, double posY) {
        return false;
    }

    @Override
    default boolean invokeMove(double posX, double posY) {
        return false;
    }

    @Override
    default boolean invokeScroll(double deltaX, double deltaY) {
        return false;
    }

}

File: Engine/src/commonMain/kotlin/org/etieskrill/engine/input/CursorInputAdapter.kt
package org.etieskrill.engine.input

interface CursorInputAdapter : CursorInputHandler {

    override fun invokeClick(button: KeyEvent, posX: Double, posY: Double): Boolean = false

    override fun invokeMove(posX: Double, posY: Double): Boolean = false

    override fun invokeScroll(deltaX: Double, deltaY: Double): Boolean = false

}
